core/serialization/deserializer/montage-reviver.js

```diff
@@ -469,14 +465,20 @@ var MontageReviver = exports.MontageReviver = Montage.specialize(/** @lends Mont
                 )) {
                 // We have a circular reference. If we wanted to forbid circular
                 // references this is where we would throw an error.
-                return Promise.resolve(this._deserializerConstructor.moduleContexts.get(location)._objects.root);
+                return this._deserializerConstructor.moduleContexts.get(location)._objects.root;
```

Comment:
shouldn't that return a promise as before when this._sync is false?

Reply:
I don't think it's necessary because `reviveMontageObject` is called by `reviveRootObject` which is called by `getObject`. `getObject` already assumes that the "object" it gets back could be a promise or a value and handles it appropriately. I think this is a case of an extra promise that we don't need.

Reply:
I don't think it's necessary because `reviveMontageObject` is called by `reviveRootObject` which is called by `getObject`. `getObject` already assumes that the "object" it gets back could be a promise or a value and handles it appropriately. I think this is a case of an extra promise that we don't need.